clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -135,4 +214,34 @@ public boolean process() {
             return true;
         }
     }
+
+    private ClusterResourceListeners configureClusterResourceListeners(
+            final Deserializer<K> keyDeserializer,
+            final Deserializer<V> valueDeserializer,
+            final List<?>... candidateLists) {
+        ClusterResourceListeners clusterResourceListeners = new ClusterResourceListeners();
+        for (List<?> candidateList: candidateLists)
+            clusterResourceListeners.maybeAddAll(candidateList);
+
+        clusterResourceListeners.maybeAdd(keyDeserializer);
+        clusterResourceListeners.maybeAdd(valueDeserializer);
+        return clusterResourceListeners;
+    }
```

Comment:
Calling instance methods from the constructor is risky. Making it static would make it safe. FYI, I think IDEA has a warning you can enable to alert you to this condition.